src/server/worker/driver/driver.go

```diff
@@ -496,6 +496,8 @@ func (d *driver) UserCodeEnv(
 	for _, input := range inputs {
 		result = append(result, fmt.Sprintf("%s=%s", input.Name, filepath.Join(d.InputDir(), input.Name, input.FileInfo.File.Path)))
 		result = append(result, fmt.Sprintf("%s_COMMIT=%s", input.Name, input.FileInfo.File.Commit.ID))
+		result = append(result, fmt.Sprintf("PACH_DATUM_%s_JOIN_ON=%s", input.Name, input.JoinOn))
+		result = append(result, fmt.Sprintf("PACH_DATUM_%s_GROUP_BY=%s", input.Name, input.GroupBy))
```

Comment:
my initial concerns with this, though I think ultimately none are necessary to address:
- we'll have these values even when they aren't used. We could skip if empty, but I suppose someone could actually rely on joining/grouping on the empty string
- there can be duplicates if there was a `Group` (multiple `common.Input` entries from the same input), but it looks like `cmd.Start()` just uses the last one
- maybe more of a concern: I don't think we do any validation on input names, and the join/group key can be basically anything. The latter is probably fine, but if the input name has `=` it would lead to the wrong variable being created. This seems unlikely, but maybe we should think about validating input names at some point.

Reply:
> we'll have these values even when they aren't used. We could skip if empty, but I suppose someone could actually rely on joining/grouping on the empty string
True, then I think we should skip setting these env variables if empty. Also I'm not sure why a user would ever want to join_on/group_by empty values.
> there can be duplicates if there was a Group (multiple common.Input entries from the same input), but it looks like cmd.Start() just uses the last one
I see... but that is fine since the values for these would always be equal right?
> Definitely. We should create a ticket to track this.